test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/LeadOnlyUndertakingSupportSpec.scala

```diff
@@ -38,36 +38,65 @@ import utils.CommonTestData.{eori3, undertaking}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class LeadOnlyUndertakingSupportSpec extends AnyWordSpecLike with MockFactory with ScalaFutures with Matchers {
+class LeadOnlyUndertakingSupportSpec extends AnyWordSpecLike with MockFactory with ScalaFutures with Matchers
+  with JourneyStoreSupport {
 
   private val mockEscService = mock[EscService]
 
   private val underTest = new FrontendController(mock[MessagesControllerComponents]) with LeadOnlyUndertakingSupport {
     override protected val escService: EscService = mockEscService
+    override protected val store: Store = mockJourneyStore
     override protected implicit val executionContext: ExecutionContext = scala.concurrent.ExecutionContext.global
   }
 
   "LeadOnlyUnderTakingSupport" should {
 
     "invoke the function" when {
 
-      "called with a request from a lead undertaking user" in {
-        mockRetrieveUndertaking(eori)(undertaking.some.toFuture)
+      def runTest() = {
+        val fakeRequest = AuthenticatedEscRequest(
+          "Foo",
+          "Bar",
+          FakeRequest(),
+          eori
+        )
+        val result = underTest.withLeadUndertaking(_ => Ok("Foo").toFuture)(fakeRequest)
+        status(result) shouldBe OK
+      }
 
-        val fakeRequest = authorisedRequestForEori(eori)
+      "called with a request from a lead undertaking user where the cache is empty" in {
+        inSequence {
+          mockGet[Undertaking](eori)(Right(Option.empty))
+          mockRetrieveUndertaking(eori)(undertaking.some.toFuture)
+          mockPut(undertaking, eori)(Right(undertaking))
+        }
 
-        val result = underTest.withLeadUndertaking(_ => Ok("Foo").toFuture)(fakeRequest)
+        runTest()
+      }
 
-        status(result) shouldBe OK
+      "called with a request from a lead undertaking user where the undertaking is cached" in {
+        inSequence {
+          mockGet[Undertaking](eori)(Right(undertaking.some))
+        }
+
+        runTest()
       }
+
     }
 
     "redirect to the account home page" when {
 
       "called with a request from a non-lead undertaking user" in {
-        mockRetrieveUndertaking(eori3)(undertaking.some.toFuture)
+        inSequence {
+          mockGet[Undertaking](eori3)(Right(undertaking.some))
+        }
 
-        val fakeRequest = authorisedRequestForEori(eori3)
+        val fakeRequest = AuthenticatedEscRequest(
```

Comment:
minor :  fakeRequest can be made as def accepting eori as param

Reply:
Ok, I'll sort this.

Reply:
Changes pushed.